### PR TITLE
TRUNK-5068 Add tests to ProgramWorkflowService

### PR DESCRIPTION
**Pull request by dilekamadushan**
https://issues.openmrs.org/browse/TRUNK-5068

## Description of what I changed
Added tests to ProgramWorkflowService

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/ProgramWorkflowServiceUnitTest.java [hunk 1]
```diff
@@ -173,4 +179,146 @@ public void saveProgram_shouldFailIfProgramWorkFlowStateTerminalIsNull() {
...
+	
+	@Test
+	public void purgePatientProgram_shouldFailForNotEmptyStatesAndTrueCascade() {
+		
+		exception.expect(APIException.class);
+		exception.expectMessage("Cascade purging of PatientPrograms is not implemented yet");
+		
+		PatientProgram patientProgram = new PatientProgram(1);
+		
+		Set<PatientState> patientStates = new Set<PatientState>() {
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/ProgramWorkflowServiceUnitTest.java]: this is not needed. you can implement this test as

```java
	@Test
	public void purgePatientProgram_shouldFailGivenNonEmptyStatesAndCascadeSetToTrue() {
		
		exception.expect(APIException.class);
		exception.expectMessage("Cascade purging of PatientPrograms is not implemented yet");	
		
		PatientProgram patientProgram = new PatientProgram();
		PatientState patientState = new PatientState();
		patientProgram.getStates().add(patientState);
		
		pws.purgePatientProgram(patientProgram, true);
	}
```

File: api/src/test/java/org/openmrs/api/ProgramWorkflowServiceUnitTest.java [hunk 2]
```diff
@@ -173,4 +179,146 @@ public void saveProgram_shouldFailIfProgramWorkFlowStateTerminalIsNull() {
...
+	}
+	
+	@Test
+	public void purgeProgram_shouldFailForNotEmptyWorkFlowsAndTrueCascade() {
+		
+		exception.expect(APIException.class);
+		exception.expectMessage("Cascade purging of Programs is not implemented yet");
+		
+		Program program = new Program(1);
+
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/ProgramWorkflowServiceUnitTest.java]: remove newlines in between your setup. Remember the arrange/act/assert. Grouping related code together makes it more readable.

```java
  ...
  exception.expectMessage("Cascade purging of Programs is not implemented yet");

  Program program = new Program(1);
  ProgramWorkflow workflow = new ProgramWorkflow(1);
  program.addWorkflow(workflow);

  pws.purgeProgram(program, true);
}
```

Comment (dilekamadushan) [reply api/src/test/java/org/openmrs/api/ProgramWorkflowServiceUnitTest.java]: @teleivo I am sorry for keep forgetting the basics

File: api/src/test/java/org/openmrs/api/ProgramWorkflowServiceUnitTest.java (hunk 1 above)
Comment (dilekamadushan) [reply api/src/test/java/org/openmrs/api/ProgramWorkflowServiceUnitTest.java]: @teleivo This is a much easier way. Thanks

**Comment by dilekamadushan** [comment]
@teleivo I modified the code according to your comments. Thank you very much for your valuable comments.

**Comment by JAT9BW98E** [comment]
No seriously thank you... I just hope you’re not a crook🌎🐺🐺🐺🐺🐺😌 Be safe and stay safe

**Comment by JAT9BW98E** [comment]
PLEASE I ONLY WANT THIS WORLD TO BE BETTER FOR ALL OF US LEAST MYSELF... pls dont be a crook but be a an asset in all levels of goodness.... lets us help people live better.... less stress and more happiness through awareness, be safe you all... hope to meet everyone soon. Love namaste🌍☠️🗿🦅🐺😌🙌🌊🌊🌊🌊🌊

**Comment by teleivo** [comment]
@dilekamadushan can you please look at the test coverage and figure out other test cases that you could write and create a new PR for it?

**Comment by dilekamadushan** [comment]
@teleivo Thank you very much for your guidance too.

**Pull request merged**